File: 47. LongestConsecutiveSequence.java
import java.util.HashSet;
import java.util.Set;

public class longestConsecutiveSequence {

    public static int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }

        int longest = 0;

        for (int num : set) {
            // Start counting only at sequence starts.
            if (!set.contains(num - 1)) {
                int current = num;
                int length = 1;

                while (set.contains(current + 1)) {
                    current++;
                    length++;
                }

                longest = Math.max(longest, length);
            }
        }

        return longest;
    }

    public static void main(String[] args) {
        System.out.println(longestConsecutive(new int[] {100, 4, 200, 1, 3, 2}));
        System.out.println(longestConsecutive(new int[] {0, 3, 7, 2, 5, 8, 4, 6, 0, 1}));
        System.out.println(longestConsecutive(new int[] {}));
    }
}

File: 47. LongestConsecutiveSequence.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;

int longestConsecutive(vector<int>& nums) {
    unordered_set<int> set(nums.begin(), nums.end());
    int longest = 0;
    for (int num : set) {
        if (!set.count(num - 1)) {
            int current = num;
            int length = 1;
            while (set.count(current + 1)) {
                current++;
                length++;
            }
            longest = max(longest, length);
        }
    }
    return longest;
}

int main() {
    vector<int> a = {100, 4, 200, 1, 3, 2};
    vector<int> b = {0, 3, 7, 2, 5, 8, 4, 6, 0, 1};
    vector<int> c = {};
    cout << longestConsecutive(a) << "\n";
    cout << longestConsecutive(b) << "\n";
    cout << longestConsecutive(c) << "\n";
    return 0;
}
